gunpowder_augmentor.py

```diff
@@ -0,0 +1,106 @@
+import zarr
+import gunpowder as gp
+import math
+import numpy as np
+import matplotlib.pyplot as plt
+import torch
+
+
+def process_zarr_data(load_path, output_shape=(30, 128, 128), stack_size=5,  device='cuda'):
+    # Open the zarr file
+    f = zarr.open(load_path, 'r')
+
+    # Define keys
+    raw = gp.ArrayKey('RAW')
+    gt = gp.ArrayKey('GROUND_TRUTH')
+
+    ########  Define paramters ########
+
+    # How many stacks to request
+    stack = gp.Stack(stack_size)
+
+    # random sampeling 
+    random_location = gp.RandomLocation()
+
+    # geometric augmentation
+    simple_augment = gp.SimpleAugment(
+    mirror_probs = [0,0,0],
+    transpose_probs = [0,0,0])
+    
+    elastic_augment = gp.ElasticAugment(
+    control_point_spacing=(30, 30, 30),
+    jitter_sigma=(1.0, 1.0, 1.0),
+    rotation_interval=(0, math.pi/2),
+    spatial_dims = 3
+    )
+
+    # signal augmentations
+    intensity_augment = gp.IntensityAugment(
+    raw,
+    scale_min=0.9,
+    scale_max=1.1,
+    shift_min=-0.01,
+    shift_max=0.01)
+
+    noise_augment = gp.NoiseAugment(raw, mode='poisson')
+
+    normalize_raw = gp.Normalize(raw)
+    normalize_gt = gp.Normalize(gt, factor=1.0/255.0)
+
+
+    ##### Gunpowder pipeline #####
+
+    # Load the data
+    source = tuple(gp.ZarrSource(
+    load_path,
```

Comment:
Improve generalizability by iterating over list of paths and fov 

Think about data structure for this input

Reply:
now loads a list of images